Modules/System/AL Rest Client/src/ALRestClient.Codeunit.al

```diff
@@ -0,0 +1,339 @@
+/// <summary>Provides functionality to easily work with the HttpClient object.</summary>
```

Comment:
[Not directly related to this PR]
Normally when we add new functionality like this, we also try to have some uptake in the application.
Not only it would bring the benefit of this module to the features that uptake it, but it would also set the example for future implementations as to how to use this functionality.

The emailing apps in this repo could be a very good place to do this if you choose to do so 🙂.

Reply:
@IhorHandziuk
Probably it's only me, but I don't get your message here. Who do you address? AJ, to change his code (then: what is the actual instruction?) or Microsoft, to plan to use the new module in near future? Or am I totally mistaken?

Reply:
@IhorHandziuk 
Some other PRs are waiting since they should use this module.
E g. The PR for a generic Microsoft Graph API wrapper.
Therefore we should get this working and nobody stops Microsoft or anyone else from refactoring any other Module afterwards.

Reply:
I'll take a look at other modules that can implement this new module. I guess the Azure Blob Storage and Azure Functions module are also candidates.